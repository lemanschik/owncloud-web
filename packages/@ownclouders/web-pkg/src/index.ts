export * from './composables'
export * from './configuration'
export * from './helpers'
export * from './http'
export * from './observer'
export * from './services'
export * from './utils'
