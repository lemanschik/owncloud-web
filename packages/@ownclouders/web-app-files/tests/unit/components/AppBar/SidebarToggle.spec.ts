import SidebarToggle from '../../../../src/components/AppBar/SidebarToggle.vue'
import { eventBus } from 'web-pkg'
import {
  createStore,
  defaultPlugins,
  mount,
  defaultStoreMockOptions,
  defaultComponentMocks
} from 'web-test-helpers'

const selectors = {
  toggleSidebarBtn: '#files-toggle-sidebar'
}

describe('SidebarToggle component', () => {
  it.each([true, false])(
    'should show the "Toggle sidebar"-button with sidebar opened and closed',
    (sideBarOpen) => {
      const { wrapper } = getWrapper({ sideBarOpen })
      expect(wrapper.find(selectors.toggleSidebarBtn).exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    }
  )
  it('publishes the toggle-event to the sidebar on click', async () => {
    const { wrapper } = getWrapper()
    const eventSpy = jest.spyOn(eventBus, 'publish')
    await wrapper.find(selectors.toggleSidebarBtn).trigger('click')
    expect(eventSpy).toHaveBeenCalled()
  })
})

function getWrapper({ sideBarOpen = false } = {}) {
  const storeOptions = { ...defaultStoreMockOptions }
  const store = createStore(storeOptions)
  const mocks = defaultComponentMocks()
  return {
    storeOptions,
    mocks,
    wrapper: mount(SidebarToggle, {
      props: { sideBarOpen },
      global: {
        mocks,
        plugins: [...defaultPlugins(), store]
      }
    })
  }
}
