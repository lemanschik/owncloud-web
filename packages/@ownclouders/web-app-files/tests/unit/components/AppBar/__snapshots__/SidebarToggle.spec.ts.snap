// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarToggle component should show the "Toggle sidebar"-button with sidebar opened and closed 1`] = `
<button aria-label="Close sidebar to hide details" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-my-s oc-p-xs" id="files-toggle-sidebar" type="button">
  <!-- @slot Content of the button -->
  <span class="oc-icon oc-icon-m oc-icon-passive">
    <!---->
  </span>
</button>
`;

exports[`SidebarToggle component should show the "Toggle sidebar"-button with sidebar opened and closed 2`] = `
<button aria-label="Open sidebar to view details" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-my-s oc-p-xs" id="files-toggle-sidebar" type="button">
  <!-- @slot Content of the button -->
  <span class="oc-icon oc-icon-m oc-icon-passive">
    <!---->
  </span>
</button>
`;
